lib/fs.js

```diff
@@ -1813,11 +1843,25 @@ fs.realpath = function realpath(path, options, callback) {
   } else if (typeof options !== 'object') {
     throw new TypeError('"options" must be a string or an object');
   }
-  callback = makeCallback(callback);
```

Comment:
I am not sure removing this line is without side effects, at least for the case that the `!nullCheck()` branch is taken.